content/docs/rfcs/37/README.md

```diff
@@ -0,0 +1,151 @@
+---
+slug: 37
+title: 37/WAKU2-NOISE-SESSIONS
+name: Session Management for Waku Noise
+status: raw
+tags: waku-core-protocol
+editor: Giuseppe <giuseppe@status.im>
+contributors: 
+---
+
+# Introduction
+
+In [35/WAKU2-NOISE](https://rfc.vac.dev/35/) we defined how Waku messages' payloads can be encrypted using key material derived from key agreements based on the [Noise Protocol Framework](http://www.noiseprotocol.org/noise.html). 
+
+Once two users complete a Noise handshake, 
+an encryption/decryption session - _or a Noise session_ - would be instantiated.
+
+This post provides an overview on how we can possibly implement and manage one or multiple Noise sessions in Waku.
+
+# Preliminaries
+
+We assume that two users, e.g. Alice and Bob, successfully completed a Noise handshake.
+
+Using [Noise terminology]((http://www.noiseprotocol.org/noise.html)), at the end of the handshake they will share:
+- two _Cipher States_ `CSOutbound` and `CSInbound`, to encrypt and decrypt outbound and inbound messages, respectively;
+- a handshake hash value `h`.
+
+As suggested in Noise specifications in regards to [Channel Binding](http://www.noiseprotocol.org/noise.html#channel-binding), 
+we can identify a Noise session with a `session-id` derived from the handshake hash value `h` shared on completion of a Noise handshake.
+
+More specifically, when Alice and Bob call [Split()](http://www.noiseprotocol.org/noise.html#the-symmetricstate-object) in order to derive the two final encryption and decryption Cipher States, 
+they further compute `session-id  = HKDF(h)` using the supported key derivation function `HKDF`.
+
+Such `session-id` will uniquely identify the Noise cryptographic session instantiated on completion of a Noise handshake,
+which would then consists of the tuple `(session-id, CSOutbound, CSInbound)`. 
+
+Once a Noise session is instantiated, 
+any further encrypted message between Alice and Bob within this session is exchanged on a `contentTopic` with name `/wakunoise/1/<ct-id>/proto`, 
+where `ct-id = Hash(Hash(session-id))`.
```

Comment:
Sure, it is to avoid an attacker to claim that he was a party in a previous session while allowing _public stale_ of sessions. To publicly stale a certain session with content topic `ct-id = Hash(Hash(session-id))`, you reveal `Hash(session-id)`, a pre-image, whose validity can publicly verified. When instantiating a new session, if you want to do channel binding with a previous one (either stale or active), you use `session-id`, which is known only to parties either if the previous session is active of was declared stale.

Reply:
>  it is to avoid an attacker to claim that he was a party in a previous session

Can you please expand on it? In which scenario would it be considered as attack? (we can also discuss in our 1:1)

Reply:
This relates to channel binding and reactivation of Noise session. The session-id allows re-activation, hence an attacker would be able to show that he was involved in a previous handshake while binding a new channel if we only consider Hash(session-id) and stale with session-id.